app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/MainActivity.kt

```diff
@@ -41,11 +54,28 @@ class MainActivity : AppCompatActivity() {
         setSupportActionBar(toolbar)
         toolbar.setupWithNavController(navController, appBarConfiguration)
 
-        findViewById<BottomNavigationView>(R.id.bottom_nav).setupWithNavController(navController)
+        val bottomNav = findViewById<BottomNavigationView>(R.id.bottom_nav)
+        bottomNav.setupWithNavController(navController)
+        val goToLearningButton = bottomNav?.menu?.get(0)
+        val goToDsEditingButton = bottomNav?.menu?.get(1)
+
+        navController.addOnDestinationChangedListener { _, _, _ ->
+            lifecycleScope.launch {
+                val user = presenter.currentUser
+                val isAdmin = user?.let { dbService.checkIsAdmin(it) }
+                if (user == null || !isAdmin!!) {
```

Comment:
`user == null` is not necessary since `isAdmin` would be null in this case. I suggest doing something like `isAdmin?.let { if (it) ....} ?: ...`

Reply:
Oh yes you're right Thank you ! Forgot to modify that. (Btw don't review all the "file changes" since a lot of those changes are from Marouane's PR.)